Tests/test_SffIO.py

```diff
@@ -368,6 +367,209 @@ def test_index2(self):
             raise ValueError("Indxing Roche/invalid_paired_E3MFGYR02.sff should fail")
 
 
+class TestSelf(unittest.TestCase):
+    """
+    These tests were originally seltests run in SffIO.py
+    """
+
+    def test_read(self):
+        filename = "Roche/E3MFGYR02_random_10_reads.sff"
+        with open(filename, "rb") as handle:
+            sff = list(SffIterator(handle))
+        with open(filename, "rb") as handle:
+            sff_trim = list(SffIterator(handle, trim=True))
+
+        from Bio import SeqIO
+        filename = "Roche/E3MFGYR02_random_10_reads_no_trim.fasta"
+        fasta_no_trim = list(SeqIO.parse(filename, "fasta"))
+        filename = "Roche/E3MFGYR02_random_10_reads_no_trim.qual"
+        qual_no_trim = list(SeqIO.parse(filename, "qual"))
+
+        filename = "Roche/E3MFGYR02_random_10_reads.fasta"
+        fasta_trim = list(SeqIO.parse(filename, "fasta"))
+        filename = "Roche/E3MFGYR02_random_10_reads.qual"
+        qual_trim = list(SeqIO.parse(filename, "qual"))
+
+        for s, sT, f, q, fT, qT in zip(sff, sff_trim, fasta_no_trim, qual_no_trim, fasta_trim, qual_trim):
+            assert s.id == f.id == q.id
+            assert str(s.seq) == str(f.seq)
+            assert s.letter_annotations["phred_quality"] == q.letter_annotations["phred_quality"]
+            assert s.id == sT.id == fT.id == qT.id
+            assert str(sT.seq) == str(fT.seq)
+            assert sT.letter_annotations["phred_quality"] == qT.letter_annotations["phred_quality"]
+
+    def test_write(self):
+        filename = "Roche/E3MFGYR02_random_10_reads.sff"
+        with open(filename, "rb") as handle:
+            metadata = ReadRocheXmlManifest(handle)
+        with open(filename, "rb") as handle:
+            sff = list(SffIterator(handle))
+        handle = BytesIO()
+        w = SffWriter(handle, xml=metadata)
+        w.write_file(sff)  # list
+        data = handle.getvalue()
+        # And again with an iterator...
+        handle = BytesIO()
+        w = SffWriter(handle, xml=metadata)
+        w.write_file(iter(sff))
+        assert data == handle.getvalue()
+        # Check 100% identical to the original:
+        filename = "Roche/E3MFGYR02_random_10_reads.sff"
+        with open(filename, "rb") as handle:
+            original = handle.read()
+            assert len(data) == len(original)
+            assert data == original
+            del data
+
+    def test_index(self):
+        filename = "Roche/greek.sff"
+        with open(filename, "rb") as handle:
+            for record in SffIterator(handle):
+                print(record.id)
+        with open(filename, "rb") as handle:
+            index1 = sorted(_sff_read_roche_index(handle))
+        with open(filename, "rb") as handle:
+            index2 = sorted(_sff_do_slow_index(handle))
+        assert index1 == index2
+
+    def test_read_wrong(self):
+        filename = "Roche/greek.sff"
+        try:
```

Comment:
Probably you can just use ``self.assertRaises(ValueError, ReadRocheXmlManifest, handle)`` here.